webofneeds/won-bot/src/main/java/won/bot/framework/eventbot/action/impl/mail/receive/MailCommandAction.java

```diff
@@ -91,22 +134,56 @@ private void processReferenceMailCommands(MimeMessage message, String referenceI
                     bus.publish(new OpenConnectionEvent(wonUri.getUri()));
                     break;
                 case IMPLICIT_OPEN_CONNECTION:
-                    bus.publish(new OpenConnectionEvent(
-                      wonUri.getUri(),  mailContentExtractor.getTextMessage(message)));
+                    bus.publish(new OpenConnectionEvent(wonUri.getUri(), mailContentExtractor.getTextMessage(message)));
                     break;
                 case SENDMESSAGE:
-                    bus.publish(new SendTextMessageOnConnectionEvent(
-                      mailContentExtractor.getTextMessage(message), wonUri.getUri()));
+                    bus.publish(new SendTextMessageOnConnectionEvent(mailContentExtractor.getTextMessage(message), wonUri.getUri()));
+                    break;
+                case CLOSE_NEED:
+                    bus.publish(new NeedDeactivatedEvent(needUri));
                     break;
-
                 case NO_ACTION:
                 default:
                     //INVALID COMMAND
                     logger.error("No command was given or assumed");
                     break;
             }
-        }catch(Exception e){
-            logger.error("no reply mail was set or found");
+        }catch (AccessControlException ace){
+            logger.error("ACCESS RESTRICTION: sender of original and command mail are not equal, command will be blocked");
+        } catch(Exception e){
+            logger.error("no reply mail was set or found: "+e.getMessage());
+        }
+    }
+
+    /**
+     * This Method tries to find a corresponding open need uri from a user(given by the from adress) and returns the
+     * corresponding need uri if there was an open need with the same title
+     * @param message used to extract sender adress and subject(title)
+     * @return
+     */
+    private URI retrieveCorrespondingNeedUriFromMailByTitle(MimeMessage message){
```

Comment:
im not sure if we should implement the taken/close use case by using the same title. what happens if there are more needs with the same title? also what happens if the message title does not exactly match the need title? 
=> i have already started to write code to take care of this problems using message ids and mailto links that the user should click on that then produce answer mails with specific subjects that hold an message id. 

E.g. when you look at the footer of some mail (e.g. match found mail) there is this:

---
close this request and do not receive mails for it anymore:<mailto:mail2won.int@gmail.com?subject=close_Message-Id_%3C1120133180.3.1487771411744.JavaMail.root%40974cfab3b863%3E>
do not publish my requests on this mailinglist and ignore mails from Web of Needs Integration Test Mail Bot in the future: <mailto:mail2won.int@gmail.com?subject=unsubscribe>

first link is for example for closing and refers the original mail by using a message id.
im not so sure how we combine this concept with closing by title ...